test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,261 @@
+        String architecture = System.getProperty("os.arch");
+
+        if (architecture.contains("x86_64") || architecture.contains("amd64")) {
+            return ArchType.X64;
+        } else if (architecture.contains("aarch64")) {
+            return ArchType.ARM64;
+        } else if (architecture.contains("arm")) {
+            return ArchType.ARM32;
+        } else if (architecture.contains("ppc64le")) {
+            return ArchType.PPC64LE;
+        } else if (architecture.endsWith("86")) {
+            return ArchType.X86;
+        } else {
+            throw new RuntimeException("Unable to detect the OS used by the host.\nOS may be unsupported.");
+        }
+    }
+
+    public static boolean enabled(Test test) {
+        OsType[] os = test.os();
+        int[] jdkVersion = test.jdkVersion();
+        JvmType[] jvm = test.jvm();
+        ArchType[] arch = test.arch();
+        
+        if (!test.enabled()) {
+            return false;
+        }
+    
+        if (os.length > 0 && !Arrays.asList(os).contains(currentOs)) {
+            return false;
+        }
+
+        if (jdkVersion.length > 0 && !Arrays.asList(jdkVersion).contains(currentJvmVersion)) {
+            return false;
+        }
+
+        if (jvm.length > 0 && !Arrays.asList(jvm).contains(currentJvm)) {
+            return false;
+        }
+
+        if (arch.length > 0 && !Arrays.asList(arch).contains(currentArch)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static Boolean run(Method m, Map skipMap, Boolean retainlogs) throws Exception{
+        Boolean passedAll = true;
+        for (Test test : m.getAnnotationsByType(Test.class)) {
+            String fullDeclaringClass = m.getDeclaringClass().getName();
+            String className = fullDeclaringClass.substring(fullDeclaringClass.lastIndexOf('.') + 1);
+
+            if (!enabled(test) || skipMap.containsKey(m.getName().toLowerCase()) || skipMap.containsKey(className.toLowerCase())) {
+                log.log(Level.FINE, "Skipped " + m.getDeclaringClass().getName() + '.' + m.getName());
+                continue;
+            }
+
+            String testName = null;
+            if (retainlogs) {
+                testName = className + '.' + m.getName();
+            }
+
+            log.log(Level.INFO, "Running " + m.getDeclaringClass().getName() + '.' + m.getName() + "...");
+
+            try (final TestProcess p = new TestProcess(test, libExt, testName);) {
+                Object holder = (m.getModifiers() & Modifier.STATIC) == 0 ? m.getDeclaringClass().newInstance() : null;
+                m.invoke(holder, p);
+                log.info("OK");
+            } catch (Throwable e) {
+                log.log(Level.FINE, e.getMessage());
+                log.log(Level.WARNING, "Test failed " + e.getCause().getMessage());
+                e.printStackTrace();
+                e.getCause().printStackTrace();
```

Comment:
Log facility should print exception stack traces. Do not call `printStackTrace` manually.